File: server/src/main/java/org/queue/cluster/Broker.scala
/*
 * Copyright 2010 LinkedIn
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *    http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

package org.queue.cluster

import org.queue.utils.Utils

/**
 * A Kafka broker
 */
private[queue] object Broker {
  def createBroker(id: Int, brokerInfoString: String): Broker = {
    val brokerInfo = brokerInfoString.split(":")
    new Broker(id, brokerInfo(0), brokerInfo(1), brokerInfo(2).toInt)
  }
}

private[queue] class Broker(val id: Int, val creatorId: String, val host: String, val port: Int) {
  
  override def toString(): String = new String("id:" + id + ",creatorId:" + creatorId + ",host:" + host + ",port:" + port)

  def getZKString(): String = new String(creatorId + ":" + host + ":" + port)
  
  override def equals(obj: Any): Boolean = {
    obj match {
      case null => false
      case n: Broker => id == n.id && host == n.host && port == n.port
      case _ => false
    }
  }
  
  override def hashCode(): Int = Utils.hashcode(id, host, port)
  
}

File: server/src/main/java/org/queue/cluster/Broker.java
package org.queue.cluster;
import org.queue.utils.Utils;
/**
 * @author gaoyvfeng
 * @ClassName Broker
 * @description:
 * @datetime 2024年 05月 21日 17:47
 * @version: 1.0
 */
public class Broker {
    private final int id;
    private final String creatorId;
    private final String host;
    private final int port;

    public Broker(int id, String creatorId, String host, int port) {
        this.id = id;
        this.creatorId = creatorId;
        this.host = host;
        this.port = port;
    }

    public String getZKString() {
        final String zkStr = new String(creatorId + ":" + host + ":" + port);
        return zkStr;
    }

    public  Broker createBroker(int id, String brokerInfoString) {
        String[] brokerInfo = brokerInfoString.split(":");
        return new Broker(id, brokerInfo[0], brokerInfo[1], Integer.parseInt(brokerInfo[2]));
    }

    @Override
    public String toString() {
        return "id:" + id + ",creatorId:" + creatorId + ",host:" + host + ",port:" + port;
    }

    @Override
    public boolean equals(Object obj) {
        if (this == obj) return true;
        if (obj == null || getClass() != obj.getClass()) return false;
        Broker broker = (Broker) obj;
        return id == broker.id && host.equals(broker.host) && port == broker.port;
    }

    @Override
    public int hashCode() {
        return Utils.hashcode(id, host, port);
    }

    public Integer getId() {
        return this.id;
    }
}
